junit/core/src/main/java/org/jboss/arquillian/junit/ArquillianTestClass.java

```diff
@@ -0,0 +1,88 @@
+package org.jboss.arquillian.junit;
+
+import org.jboss.arquillian.test.spi.LifecycleMethodExecutor;
+import org.jboss.arquillian.test.spi.TestRunnerAdaptor;
+import org.jboss.arquillian.test.spi.TestRunnerAdaptorBuilder;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+/**
+ * Class rule for Arquillian tests. Allows arquillian to be combined with other runners.
+ * Always use both rules together to get the full functionality of Arquillian.
+ * <p>
+ * <pre>
+ * @ClassRule
+ * public static ArquillianTestClass arquillianTestClass = new ArquillianTestClass();
+ * @Rule
+ * public ArquillianTest arquillianTest = new ArquillianTest();
+ * </pre>
+ *
+ * @author <a href="mailto:alexander.schwartz@gmx.net">Alexander Schwartz</a>
+ */
+public class ArquillianTestClass implements TestRule {
+
+    private TestRunnerAdaptor adaptor;
+
+    public Statement apply(final Statement base, final Description description) {
+        return new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+                State.runnerStarted();
+                // first time we're being initialized
+                if (!State.hasTestAdaptor()) {
+                    // no, initialization has been attempted before and failed, refuse
+                    // to do anything else
+                    if (State.hasInitializationException()) {
+                        // failed on suite level, ignore children
+                        // notifier.fireTestIgnored(getDescription());
+                        throw new RuntimeException(
+                                "Arquillian has previously been attempted initialized, but failed. "
+                                        + "See cause for previous exception",
```

Comment:
I'm still not sure if that is proper English, but in other places it's

> Arquillian has previously been attempted initialized, but failed. See previous exceptions for cause.

Reply:
I've seen this message before and it confused the hell out of me, after reading the surround code I think the following explains it better:

'Arquillian initialization has already been attempted, but failed. See previous exceptions for cause.'

Reply:
![agree](https://user-images.githubusercontent.com/719616/34178226-b8fd3e9a-e506-11e7-941e-f6e59063b0b2.gif)

Thanks :)

@dipak-pawar as you are on it - can you change it also in [`Arquillian.java`](https://github.com/arquillian/arquillian-core/blob/master/junit/core/src/main/java/org/jboss/arquillian/junit/Arquillian.java#L96)

Reply:
sure